src/main/java/frc/robot/subsystems/Feeder.java

```diff
@@ -13,18 +14,33 @@
 import frc.robot.utils.diag.DiagColorSensor;
 import frc.robot.utils.smartshuffleboard.SmartShuffleboard;
 
+import java.util.ArrayList;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+
 public class Feeder extends SubsystemBase {
 
     private final WPI_TalonSRX feederMotor;
     private final I2C.Port i2cPort = I2C.Port.kMXP;
     private final ColorSensor colorSensor;
+    private final BlockingQueue<ColorMatchResult> colorReadingBuffer = new LinkedBlockingQueue<>();
+    private final ArrayList<ColorMatchResult> resultsSinceLastTic = new ArrayList<>();
+
 
     public Feeder() {
         this.feederMotor = new WPI_TalonSRX(Constants.FEEDER_MOTOR_ID);
         this.feederMotor.setNeutralMode(NeutralMode.Brake);
-
         colorSensor = new ColorSensor(i2cPort);
         Robot.getDiagnostics().addDiagnosable(new DiagColorSensor("Feeder", "Color Sensor", colorSensor));
+        new Thread(() -> {
+            try {
+                while (true) {
```

Comment:
A tight loop like that would kill the CPU. Need to insert a sleep to reduce the frequency of polling to the desired one.

Reply:
done